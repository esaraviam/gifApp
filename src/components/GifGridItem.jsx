import React from 'react'
import PropTypes from 'prop-types'

import {Card , CardContent , CardMedia} from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles({
    root: {
      minWidth: 300,
      maxWidth: 300,
      margin: 10,
    },
    animate__animated:{},
    animate__fadeInLeft:{}
  });

const GifGridItem = ({title , url}) => {
    const classes = useStyles();

    return (
        <Card className={ classes.root} >
            <CardMedia 
            component="img"
            image={url}
            title={title}
            height="140"
            /> 
            <CardContent>
                <p>{title}</p>
            </CardContent>
        </Card>
        // <div className="card animate__animated animate__fadeInLeft ">
        //     <img src={url} alt={title} />  
        //     <p>{title}</p>
        // </div>
    )
}

GifGridItem.propTypes = {
    title:PropTypes.string.isRequired,
    url:PropTypes.string.isRequired
}

export default GifGridItem
