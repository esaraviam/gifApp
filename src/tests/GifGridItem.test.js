import GifGridItem from '../components/GifGridItem'
import React from 'react'
import { shallow } from 'enzyme'

describe('Test GifGridItem Component', () => {
            const title = 'Test'
            const url = 'https://media3.giphy.com/'
            const wrapper = shallow( < GifGridItem title = { title }
                url = { url }
                / > )
                test('should load the component Propperly', () => {

                    expect(wrapper).toMatchSnapshot()
                }) 
                
                test('should load the title properly', () => {

                    const p = wrapper.find('p')


                    expect(p.text().trim()).toBe('Test')
                }) 
                
                test('should load image', () => {

                    const img = wrapper.find('img')
                    const source = img.prop('src')
                    const alt = img.prop('alt')

                    expect(source).toBe(url)
                    expect(alt).toBe(title)
                })

                {/* test('should have the class animate__fadeInLeft' , () => {
                    const div = wrapper.find('div')
                    
                    expect(div.hasClass('animate__fadeInLeft')).toBe(true)
                    

                }) */}



            })