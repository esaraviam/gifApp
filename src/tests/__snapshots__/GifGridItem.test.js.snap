// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GifGridItem Component should load the component Propperly 1`] = `
<WithStyles(ForwardRef(Card))
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(CardMedia))
    component="img"
    height="140"
    image="https://media3.giphy.com/"
    title="Test"
  />
  <WithStyles(ForwardRef(CardContent))>
    <p>
      Test
    </p>
  </WithStyles(ForwardRef(CardContent))>
</WithStyles(ForwardRef(Card))>
`;
